packages/react-native/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModuleManager.mm

```diff
@@ -404,7 +404,10 @@ - (instancetype)initWithBridgeProxy:(RCTBridgeProxy *)bridgeProxy
     }
     _turboModuleCache.insert({moduleName, turboModule});
 
-    if ([module respondsToSelector:@selector(installJSIBindingsWithRuntime:)]) {
+    if ([module respondsToSelector:@selector(installJSIBindingsWithRuntime:callInvoker:)]) {
```

Comment:
so we do have an API to init the TM with the call invoker, i'm wondering if we can use this and then reference it in the implementation of installJSIBindingsWithRuntime:

https://github.com/facebook/react-native/blob/main/packages/react-native/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModuleManager.mm#L693-L697

Reply:
I mean we don't have the jsi::Runtime in there where you linked - or are you saying a TurboModule can just fetch the `CallInvoker` themselves using the `callInvoker` property (if implemented)? I guess that's fine too, but I think this API is a bit more verbose

Reply:
I think both of these options work, and perhaps for symmetry with Android, sharing the callInvoker here is nice too? But it would allow us to avoid a breaking API change here too, which is nice.

Reply:
> so we do have an API to init the TM with the call invoker, i'm wondering if we can use this and then reference it in the implementation of installJSIBindingsWithRuntime:
> 
> https://github.com/facebook/react-native/blob/main/packages/react-native/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModuleManager.mm#L693-L697

What's the difference between this codepath and the one in https://github.com/facebook/react-native/blob/main/packages/react-native/React/Base/RCTModuleData.mm#L206-L208 ? Legacy native modules vs TurboModules? Or old bridge vs bridgeless?

Reply:
The one in RCTModuleData can not access jsi::Runtime - it both doesn't have a reference to it, and it isn't guaranteed to run on the proper Thread (legacy modules)

Reply:
> > so we do have an API to init the TM with the call invoker, i'm wondering if we can use this and then reference it in the implementation of installJSIBindingsWithRuntime:
> > https://github.com/facebook/react-native/blob/main/packages/react-native/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModuleManager.mm#L693-L697
> 
> What's the difference between this codepath and the one in https://github.com/facebook/react-native/blob/main/packages/react-native/React/Base/RCTModuleData.mm#L206-L208 ? Legacy native modules vs TurboModules? Or old bridge vs bridgeless?

it's just a forward compatible API. so the idea is that you can change your code and not worry about legacy vs new arch, you will be guaranteed that your module has access to the callinvoker.